trouper.py

```diff
@@ -0,0 +1,51 @@
+class Number(int):
+
+    def __init__(self, num):
+        int.__init__(self)
+        self.num = num
+
+    # Utility Function
+    def find_divider_by_index(self, index, current_index, current_divider):
+        while current_index < index:
+            current_divider += 1
+            if current_divider > self.num:
+                return None, None
+            if self.num % current_divider == 0:
+                current_index += 1
+        return current_index, current_divider
+
+    # Utility Function
+    def find_dividers_by_slice(self, start_index, end_index):
+        result = []
+        current_divider = 1
+        current_index = 0
+        index = start_index
+        while index < end_index:
+            current_index, current_divider = self.find_divider_by_index(
+                index, current_index, current_divider)
+            if current_divider is None:
+                return result
+            result.append(current_divider)
+            index += 1
+        return result
+
+    def __getitem__(self, index):
+        if type(index) == int:
+            index, divider = self.find_divider_by_index(index, 0, 1)

```

Comment:
Why return `index`?

Reply:
find_divider_by_index returns two elements: index, divider (because it is also used by find_dividers_by_slice). I converted the above mentioned line to fetch only the second element (the divider).